src/colmap/estimators/bundle_adjustment.cc

```diff
@@ -825,6 +828,189 @@ void RigBundleAdjuster::ParameterizeCameraRigs(Reconstruction* reconstruction) {
   }
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// PositionPriorBundleAdjuster
+////////////////////////////////////////////////////////////////////////////////
+
+PositionPriorBundleAdjuster::PositionPriorBundleAdjuster(
+    const BundleAdjustmentOptions& options,
+    const BundleAdjustmentConfig& config)
+    : BundleAdjuster(options, config) {
+  prior_options_.prior_position_covariance =
+      options_.prior_position_std.cwiseAbs2().asDiagonal();
+}
+
+bool PositionPriorBundleAdjuster::Solve(Reconstruction* reconstruction) {
+  loss_function_ =
+      std::unique_ptr<ceres::LossFunction>(options_.CreateLossFunction());
+
+  if (options_.use_robust_loss_on_prior_position) {
+    prior_loss_function_ =
+        std::make_unique<ceres::CauchyLoss>(options_.prior_position_loss_scale);
+  }
+
+  // Compute initial squared error between position priors & current images
+  // projection center and prepare data for RANSAC-based sim3 alignment
+  std::vector<double> vini_err2_wrt_prior;
+  std::vector<Eigen::Vector3d> v_src, v_dst;
+  vini_err2_wrt_prior.reserve(config_.NumImages());
+  v_src.reserve(config_.NumImages());
+  v_dst.reserve(config_.NumImages());
+  for (const auto& image_id : config_.Images()) {
+    const auto& image = reconstruction->Image(image_id);
+    if (image.HasPosePrior()) {
+      v_src.push_back(image.ProjectionCenter());
+      v_dst.push_back(image.WorldFromCamPrior().position);
+      vini_err2_wrt_prior.push_back(
+          (v_src.back() - v_dst.back()).squaredNorm());
+    }
+  }
+
+  // Fallback to classic BA if not enough registered images with a prior
+  // position
+  prior_options_.use_prior_position = v_src.size() > 2;
+
+  // Fix 7-DOFs of BA problem for classic BA
+  if (!prior_options_.use_prior_position) {
+    const std::vector<image_t>& reg_image_ids = reconstruction->RegImageIds();
+    config_.SetConstantCamPose(reg_image_ids[0]);
+    config_.SetConstantCamPositions(reg_image_ids[1], {0});
+  }
+
+  // Apply RANSAC-based Sim3 Alignment
+  if (prior_options_.use_prior_position) {
+    LOG(INFO) << "Initial alignment error w.r.t. prior position:";
```

Comment:
nit: Shall we change these LOG(INFO) to VLOG(2) here and below in the same scope?

Reply:
+1

Reply:
Done :+1: